tests/functional/cli/05-colour.t

```diff
@@ -23,16 +23,19 @@
 if [[ "$OSTYPE" != "linux-gnu"* ]]; then
     skip_all "Tests not compatibile with $OSTYPE"
 fi
-set_test_number 8
+set_test_number 7
 
 ANSI='\e\['
 
 # No redirection.
 script -q -c "cylc scan -t rich" log > /dev/null 2>&1
 grep_ok "$ANSI" log -P  # color
 
-script -q -c "cylc scan -t rich --color=never" log > /dev/null 2>&1
-grep_fail "$ANSI" log -P  # no color
+# FIXME: this test doesn't work because of some interaction between the
+# `script` command and colorama. The output includes a color reset char at
+# the end despite using colorama.init(autoreset=False).
+# script -q -c "cylc scan -t rich --color=never" log > /dev/null 2>&1
+# grep_fail "$ANSI" log -P  # no color
```

Comment:
This test wasn't testing what it was supposed to test before?

Reply:
Ok, we should link an issue to this FIXME.

Would this close it: https://github.com/cylc/cylc-flow/issues/6076 ?

Reply:
No (I tried turning autoreset back on even though that wouldn't make any sense for it to fix the test, and it didn't help).

---

N.B. I had a look at how colorama works. It bills itself as a package meant only for making colour work on Windows, but I guess we are using it for its `init(strip=not use_color)` function that can automatically strip ANSI color chars from `print()` calls (it actually patches `sys.stdout` and `sys.stderr`). However there are >100 open issues and it is in a call-for-maintainers state: https://github.com/tartley/colorama/issues/300.

Reply:
Link to #6467 added